core/src/main/scala/iliad/gl/graphModel.scala

```diff
@@ -0,0 +1,413 @@
+package iliad
+package gl
+
+import iliad.syntax.all._
+import iliad.std.list._
+
+import cats._
+import cats.data._
+import cats.implicits._
+
+import quiver.{LNode, LEdge, Decomp}
+import QuiverExtra._
+
+object GraphModel {
+
+  object Output {
+    sealed trait Constructor
+    sealed trait Instance
+  }
+
+  object Texture {
+    sealed trait Uniform
+    case class Constructor(name: String,
+                           format: gl.Texture.Format,
+                           viewport: Vec2i,
+                           isDouble: Boolean)
+        extends Output.Constructor
+    case class Instance(name: String, constructor: Constructor)
+        extends Uniform
+        with Output.Instance
+
+    case class Image(name: String, format: gl.Texture.Format, viewport: Vec2i)
+        extends Uniform
+  }
+
+  object Renderbuffer {
+    case class Constructor(name: String,
+                           format: RenderbufferInternalFormat,
+                           viewport: Vec2i)
+        extends Output.Constructor
+    case class Instance(name: String, constructor: Constructor)
+        extends Output.Instance
+  }
+
+  object Framebuffer {
+    sealed trait Constructor
+    sealed trait Instance
+
+    case object OnScreen extends Constructor with Instance
+
+    case class OffScreenConstructor(
+        buffers: List[(FramebufferAttachment, Output.Constructor)])
+        extends Constructor {
+      def textures: List[Texture.Constructor] =
+        buffers.map(_._2).filterClass[Texture.Constructor]
+    }
+
+    case class OffScreenInstance(
+        instances: List[(FramebufferAttachment, Output.Instance)])
+        extends Instance {
+
+      val hasDoubleTexture: Boolean = instances
+        .map(_._2)
+        .filterClass[Texture.Instance]
+        .exists(_.constructor.isDouble)
+    }
+  }
+
+  object Model {
+    case class Constructor(name: String)
+    case class Instance(name: String,
+                        constructor: Constructor,
+                        model: gl.Model)
+  }
+
+  sealed trait Node
+  object Node {
+    sealed trait Constructor {
+      def name: String
+      def framebuffer: Framebuffer.Constructor
+      def lNode: LNode[Constructor, String] = LNode(this, name)
+    }
+    sealed trait Instance {
+      def name: String
+      def constructor: Constructor
+      def lNode: LNode[Instance, String] = LNode(this, name)
+    }
+  }
+
+  object Draw {
+    case class Constructor(
+        name: String,
+        program: Program.Unlinked,
+        primitive: PrimitiveType,
+        capabilities: Set[Capability],
+        colorMask: ColorMask,
+        isInstanced: Boolean,
+        model: Model.Constructor,
+        framebuffer: Framebuffer.Constructor
+    ) extends Node.Constructor
+
+    case class Piped(
```

Comment:
This won't work as soon as we add valves, because there will be multiple uniforms to a texture.
Perhaps this should be contained solely in the links?